### PR TITLE
Add working_directory parameter

### PR DESCRIPTION
**Pull request by jmgueirard**
* Add working_directory parameter in order to be able to use the orb in
mono repo project

I reworked the way we are handling steps, by checking if the `working_directory` parameters is present and executing the same commands. The drawback is that you can't specify custom command if using `working_directory` parameters
Circleci doesn't allow to use parameters inside another parameters which lead me to this solution

### CODE REVIEW

**Comment by jmgueirard** [comment]
As discussed with @FrancoisPoinsot, we could also add a condition if working_directory is set and navigate to the specified folder
That will avoid to have duplicate steps

File: src/jobs/build_push_image.yml
Comment (robert-nemet) [inline src/jobs/build_push_image.yml:102]: This as well can be separate command with working dir parameter. As I suggested for `quality_gate` command. Then, it can be reused for default values as well.

File: src/jobs/deploy_job.yml
Comment (robert-nemet) [inline src/jobs/deploy_job.yml:56]: Jobs already have working_directory parameter. See: https://circleci.com/docs/2.0/configuration-reference/#jobs
Rename it to void conflict.

File: src/jobs/quality_gate_job.yml
Comment (robert-nemet) [inline src/jobs/quality_gate_job.yml:19]: Jobs already have working_directory parameter. See: https://circleci.com/docs/2.0/configuration-reference/#jobs
Rename it to void conflict.

File: src/jobs/quality_gate_job.yml
Comment (robert-nemet) [inline src/jobs/quality_gate_job.yml:58]: Comparing this job and command `quality_gate` I think we should reuse it and make this job simpler.

**Review by robert-nemet — CHANGES_REQUESTED** [review]
Before merging we should test changes. We can do this by publishing branch and taking published version to build/deploy app.

**Review by michaelkuechler — COMMENTED** [review]
@jmgueirard i guess we should have a look at this in a pairing session considering the changes i did in #17

**Pull request closed without merging**
